src/sentry/snuba/metrics/utils.py

```diff
@@ -0,0 +1,143 @@
+__all__ = (
+    "MAX_POINTS",
+    "GRANULARITY",
+    "TS_COL_QUERY",
+    "TS_COL_GROUP",
+    "FIELD_REGEX",
+    "TAG_REGEX",
+    "MetricOperation",
+    "MetricUnit",
+    "MetricType",
+    "OP_TO_SNUBA_FUNCTION",
+    "AVAILABLE_OPERATIONS",
+    "OPERATIONS_TO_ENTITY",
+    "METRIC_TYPE_TO_ENTITY",
+    "ALLOWED_GROUPBY_COLUMNS",
+    "Tag",
+    "TagValue",
+    "MetricMeta",
+    "MetricMetaWithTagKeys",
+    "OPERATIONS",
+    "DEFAULT_AGGREGATES",
+    "UNIT_TO_TYPE",
+    "DerivedMetricParseException",
+    "TimeRange",
+)
+
+
+import re
+from datetime import datetime
+from typing import Collection, Literal, Mapping, Optional, Protocol, Sequence, TypedDict
+
+from sentry.snuba.dataset import EntityKey
+
+MAX_POINTS = 10000
+GRANULARITY = 24 * 60 * 60
+TS_COL_QUERY = "timestamp"
+TS_COL_GROUP = "bucketed_time"
+
+#: Max number of data points per time series:
+# ToDo modify this regex to only support the operations provided
+FIELD_REGEX = re.compile(r"^(\w+)\(((\w|\.|_)+)\)$")
+TAG_REGEX = re.compile(r"^(\w|\.|_)+$")
+
+#: A function that can be applied to a metric
+MetricOperation = Literal["avg", "count", "max", "min", "p50", "p75", "p90", "p95", "p99"]
```

Comment:
Could we create a definition of all metric operations with full type signatures, for instance `avg(distribution) -> scalar` (logically)? Even if we're not exposing this via the API, we'll need it to construct derived metrics.

Reply:
Yes